test/e2e/nodeconditions.go

```diff
@@ -0,0 +1,249 @@
+		fillDiskSpace(c)
+	})
+
+	AfterEach(func() {
+		defer framework.afterEach()
+		recoverDiskSpace(c)
+	})
+
+	It("runs out of disk space", func() {
+		// Get all nodes, and kick off the test on each.
+		nodelist, err := listNodes(c, labels.Everything(), fields.Everything())
+		if err != nil {
+			Failf("Error getting nodes: %v", err)
+		}
+		unfilledNode := nodelist.Items[unfilledNodeIdx]
+		By(fmt.Sprintf("Get CPU capacity on node %s", unfilledNode.Name))
+
+		milliCPU := unfilledNode.Status.Capacity.Cpu().MilliValue()
+		// Per pod CPU should be just enough to fit only (numNodeConditionsPods - 1) pods on the
+		// given node. We compute this value by dividing the available CPU capacity on the given
+		// node by (numNodeConditionsPods - 1) and subtracting ϵ from it.
+		podCPU := (milliCPU / (numNodeConditionsPods - 1)) - (milliCPU / 10)
+
+		ns := framework.Namespace.Name
+		podClient := c.Pods(ns)
+
+		By("Creating pods")
+
+		podIds := make([]podId, 0)
+
+		for i := 0; i < numNodeConditionsPods; i++ {
+			id := podId{
+				name:  fmt.Sprintf("pod-nodecondition-%d", i),
+				value: strconv.Itoa(time.Now().Nanosecond()),
+			}
+			createPod(c, ns, id, podCPU)
+			podIds = append(podIds, id)
+		}
+		defer deletePods(c, ns)
+
+		By("Waiting for all but one pods to be scheduled")
+		// Wait for all pods except the last one to be running
+		for _, id := range podIds[:len(podIds)-1] {
+			expectNoError(framework.WaitForPodRunning(id.name))
+			pods, err := podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": id.value})), fields.Everything())
+			expectNoError(err)
+			for _, pod := range pods.Items {
+				Expect(pod.Spec.NodeName).ToNot(Equal(""))
+			}
+		}
+
+		By(fmt.Sprintf("Finding a failed scheduler event for pod %s", podIds[len(podIds)-1].name))
+		schedEvents, err := c.Events(ns).List(
+			labels.Everything(),
+			fields.Set{
+				"involvedObject.kind":      "Pod",
+				"involvedObject.name":      podIds[len(podIds)-1].name,
+				"involvedObject.namespace": ns,
+				"source":                   "scheduler",
+				"reason":                   "FailedScheduling",
+			}.AsSelector())
+		expectNoError(err)
+
+		Expect(schedEvents.Items).ToNot(BeEmpty())
+	})
+})
+
+func createPod(c *client.Client, ns string, id podId, milliCPU int64) {
+	podClient := c.Pods(ns)
+
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name: id.name,
+			Labels: map[string]string{
+				"name": "foo",
+				"time": id.value,
+			},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  "nginx",
+					Image: "beta.gcr.io/google_containers/pause:2.0",
+					Resources: api.ResourceRequirements{
+						Requests: api.ResourceList{
+							// Request enough CPU to fit only two pods on a given node.
+							api.ResourceCPU:     *resource.NewMilliQuantity(milliCPU, resource.DecimalSI),
+							api.ResourceStorage: *resource.NewQuantity(50*gb, resource.DecimalSI),
```

Comment:
Is there a reason you have this? We don't do anything with storage resource currently AFAIK. I would just delete it to avoid confusing someone who might read this test.